src/sentry/api/endpoints/organization_teams.py

```diff
@@ -142,25 +143,53 @@ def post(self, request: Request, organization, **kwargs) -> Response:
 
         :pparam string organization_slug: the slug of the organization the
                                           team should be created for.
-        :param string name: the optional name of the team.
-        :param string slug: the optional slug for this team.  If
-                            not provided it will be auto generated from the
-                            name.
+        :qparam string name: the optional name of the team.
+        :qparam string slug: the optional slug for this team. If not provided it will be auto
+                             generated from the name.
+        :qparam bool set_admin: If this is true, the user is added to the as a Team Admin
+                                instead of regular member
         :auth: required
         """
         serializer = TeamPostSerializer(data=request.data)
 
-        if serializer.is_valid():
-            result = serializer.validated_data
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+        result = serializer.validated_data
+        set_admin = result.get("set_admin")
+
+        if set_admin:
+            if not features.has("organizations:team-roles", organization):
+                return Response(
+                    {
+                        "detail": "Unable to set user as Team Admin because organization:team-roles flag is not enabled"
```

Comment:
we have existing behavior where we set the team-role to `None` when the org doesn't have the flag https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_member/details.py#L202-L207

Reply:
I think the problem with ignoring the flag is that someone who isn't a Team Admin on any team requires being set as one if they want to create a new project. When they try to create a project it'll work but they didn't have the permission to do so. We could feature flag the `Create Project` button behind `team-roles` and `team-project-creation-all` flag as a potential workaround, but there might be a race condition of someone turning off the flags but having clicked create project already and then sending the POST

Reply:
although since you're requiring both flags, you could also lump them together in one if statement

Reply:
ok I'll change it to `You do not have permission to join a new team as a team admin` and combine the ifs